storage/client_split_test.go

```diff
@@ -76,7 +76,9 @@ func TestStoreRangeSplitAtIllegalKeys(t *testing.T) {
 		keys.Meta1Prefix,
 		keys.MakeKey(keys.Meta1Prefix, []byte("a")),
 		keys.MakeKey(keys.Meta1Prefix, proto.KeyMax),
+		keys.Meta2KeyMax,
```

Comment:
interesting. there's a client-side test for it (but not the other illegal split keys) in `kv/dist_sender_server_test.go`. I can either remove it entirely, or make that test more comprehensive.